File: src/main/java/level_a/java/BeautifulMatrix.java
package level_a.java;

import java.io.*;

public class BeautifulMatrix {
    public static void main(String[] args) throws IOException {
        String data = "0 0 0 0 0\n" +
                "0 0 0 0 1\n" +
                "0 0 0 0 0\n" +
                "0 0 0 0 0\n" +
                "0 0 0 0 0";
        InputStream in = new ByteArrayInputStream(data.getBytes("UTF-8"));
        System.setIn(in);
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int [] pos = new int[2];
        for(int i=1; i<=5; i++){
            String[] line = br.readLine().split("\\s+");
            for(int j=0; j<5; j++) {
                if (line[j].equals("1")) {
                    pos = new int[]{i, j+1};
                    break;
                }
            }
        }

        int moves = 0;
        moves += Math.abs(pos[0]-3);
        moves += Math.abs(pos[1]-3);
        System.out.print(moves);
    }
}

File: src/main/java/level_a/kotlin/BeautifulMatrix.kt
package level_a.kotlin

import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.abs

fun main(args: Array<String>){
    val br = BufferedReader(InputStreamReader(System.`in`))
    var row = 0
    var col = 0
    for(i in 0..4){
        val inputRow = br.readLine().split("\\s+".toRegex())
        for(j in 0..4){
            if(inputRow[j] == "1"){
                row = i
                col = j
            }
        }
    }
    print(abs(row-2) + abs(col-2))
}